packages/@react-spectrum/overlays/src/Popover.tsx

```diff
@@ -102,6 +102,38 @@ const PopoverWrapper = forwardRef((props: PopoverWrapperProps, ref: RefObject<HT
   let {modalProps} = useModal({
     isDisabled: isNonModal
   });
+  let [isHideArrow, setIsHideArrow] = useState(hideArrow);
+
+  useLayoutEffect(() => {
+    // check position of arrow and popover boundries
+    if (ref.current && arrowProps) {
+      let propsHideArrow = hideArrow;
+
+      // figure out the location of the offset and which popover dimension to use
+      let offset = arrowProps.style.left ? 'left' : 'top';
+      let overlayCrossSize = offset === 'left' ? ref.current.offsetWidth : ref.current.offsetHeight;
+
+      // for the start and top, no change is made to arrowPosition not caught in a condiational below
+      if (12 > arrowProps.style[offset] && arrowProps.style[offset] > 0) {
```

Comment:
Where does this 12 come from? It maybe a good idea to store it in a descriptive var or leave a comment that explains it

Reply:
The 12 comes from https://github.com/adobe/react-spectrum/blob/d9f9c3bec438b9308d7fdb242ba230efdb973f9f/packages/%40react-aria/overlays/src/useOverlayPosition.ts#L78 I recall there being a conversation about not needing to make a static declaration in react-aria for this padding, but I can't find it. Since the 12 now crosses packages it makes sense to declare.